examples/ibm-power/main.tf

```diff
@@ -1,31 +1,43 @@
-data "ibm_pi_image" "data_source_image" {
-  pi_cloud_instance_id = var.cloud_instance_id
-  pi_image_name        = var.image_name
+# Create a workspace
+resource "ibm_resource_instance" "location" {
+  name              = var.workspace_name
+  resource_group_id = var.resource_group_id
+  location          = var.datacenter
+  service           = "power-iaas"
+  plan              = "power-virtual-server-group"
 }
-resource "ibm_pi_key" "key" {
-  pi_cloud_instance_id = var.cloud_instance_id
-  pi_key_name          = var.ssh_key_name
-  pi_ssh_key           = var.ssh_key_rsa
+
+# Create an image
+resource "ibm_pi_image" "image" {
+  pi_cloud_instance_id = ibm_resource_instance.location.guid
+  pi_image_name        = var.image_name
+  pi_image_id          = var.image_id
 }
-data "ibm_pi_key" "data_source_key" {
-  depends_on = [ibm_pi_key.key]
+data "ibm_pi_image" "data_source_image" {
+  depends_on = [ibm_pi_image.image]
 
-  pi_cloud_instance_id = var.cloud_instance_id
-  pi_key_name          = var.ssh_key_name
+  pi_cloud_instance_id = ibm_resource_instance.location.guid
+  pi_image_name        = var.image_name
```

Comment:
```suggestion
  pi_image_name        = data.ibm_pi_image.pi_image_name
```

Reply:
Changed code to remove depends on and use the resources to let terraform figure out the dependencies.